packages/contracts/contracts/AggregatorWithIPCreationProxy.sol

```diff
@@ -0,0 +1,24 @@
+pragma solidity ^0.5.0;
+pragma experimental ABIEncoderV2;
+
+/* External Imports */
+import "openzeppelin-solidity/contracts/token/ERC20/ERC20.sol";
+
+/* Internal Imports */
+import { Aggregator } from "./Aggregator.sol";
+import { PlasmaRegistry } from "./PlasmaRegistry.sol";
+
+contract AggregatorWithIPCreationProxy {
+  address payable public owner;
+  PlasmaRegistry public plasmaRegistry;
+
+  constructor(PlasmaRegistry _plasmaRegistry, address _authenticationAddress, string memory data) public {
+    owner = msg.sender;
+    plasmaRegistry = _plasmaRegistry;
+    plasmaRegistry.addAggregator(_authenticationAddress);
+  }
+
+  function deleteThisContract() public {
+    selfdestruct(owner);
+  }
+}
```

Comment:
Yes, it's possible to add selfdestruct(...) to the constructor. The reason why I chose to create a separate function by design is in case we want to access the proxy contract / for future testing purposes.

Reply:
> Also, I believe that the address passed to selfdestruct sends the smart contract balance to the owner, which in this case is always going to be zero. So we can probably just send it to the zero address to avoid the SSTORE of storing the owner.
Correct - will make the change.